File: Source/Back-end/src/main/java/org/elis/manutenzione/dto/response/GetStanzeResponse.java
package org.elis.manutenzione.dto.response;

import lombok.Data;

/**
 * DTO che rappresenta la risposta alla richiesta di ottenere la lista delle stanze.
 */
@Data
public class GetStanzeResponse {

    /**
     * L'ID della stanza.
     */
    private long id;

    /**
     * Il nome della stanza.
     */
    private String nome;

    /**
     * Il nucleo a cui appartiene la stanza.
     */
    private String nucleo;

    /**
     * Il piano in cui si trova la stanza.
     */
    private int piano;

    /**
     * La capienza massima della stanza.
     */
    private int capienza;

    /**
     * Il numero di residenti attualmente presenti nella stanza.
     */
    private int residenti;
}

File: Source/Front-end/front-end/src/app/model/response/get-stanze-response.ts
import {Tipo} from "../models/tipo";

export class GetStanzeResponse {
  id: number;
  nome: string;
  nucleo: string;
  piano: number;
  capienza: number;
  residenti: number;

  constructor(id: number, nome: string, nucleo: string, piano: number, capienza: number, residenti: number) {
    this.id = id;
    this.nome = nome;
    this.nucleo = nucleo;
    this.piano = piano;
    this.capienza = capienza;
    this.residenti = residenti;
  }
}
